python/plugins/processing/algs/gdal/contour.py

```diff
@@ -74,6 +75,10 @@ def initAlgorithm(self, config=None):
         self.addParameter(QgsProcessingParameterString(self.FIELD_NAME,
                                                        self.tr('Attribute name (if not set, no elevation attribute is attached)'),
                                                        defaultValue='ELEV',
+                                                       optional=True))       
+        self.addParameter(QgsProcessingParameterString(self.OPTIONS,
```

Comment:
This should be marked as an "advanced" option. See 

https://github.com/qgis/QGIS/blob/master/python/plugins/processing/algs/gdal/ClipVectorByMask.py#L58

You need the whole block:

        options_param = QgsProcessingParameterString(self.OPTIONS,
                                                     self.tr('Additional creation options'),
                                                     defaultValue='',
                                                     optional=True)
        options_param.setFlags(options_param.flags() | QgsProcessingParameterDefinition.FlagAdvanced)
        self.addParameter(options_param)